recipes-sota/aktualizr/aktualizr_git.bb

```diff
@@ -32,29 +27,34 @@ BRANCH ?= "master"
 
 S = "${WORKDIR}/git"
 
-inherit cmake
-
-inherit systemd
+inherit pkgconfig cmake systemd
 
 SYSTEMD_PACKAGES = "${PN} ${PN}-secondary"
 SYSTEMD_SERVICE_${PN} = "aktualizr.service"
 SYSTEMD_SERVICE_${PN}-secondary = "aktualizr-secondary.socket"
 
-BBCLASSEXTEND =+ "native"
-
-require garage-sign-version.inc
-
-EXTRA_OECMAKE = "-DWARNING_AS_ERROR=OFF \
-                 -DCMAKE_BUILD_TYPE=Release \
-                 -DAKTUALIZR_VERSION=${PV} \
-                 -DBUILD_LOAD_TESTS=OFF"
-EXTRA_OECMAKE_append_class-target = " -DBUILD_OSTREE=ON \
-                                      ${@bb.utils.contains('SOTA_CLIENT_FEATURES', 'hsm', '-DBUILD_P11=ON', '', d)} "
-EXTRA_OECMAKE_append_class-native = " -DBUILD_SOTA_TOOLS=ON \
-                                      -DBUILD_OSTREE=OFF \
-                                      -DBUILD_SYSTEMD=OFF \
-                                      -DGARAGE_SIGN_VERSION=${GARAGE_SIGN_VERSION} \
-                                      -DGARAGE_SIGN_SHA256=${GARAGE_SIGN_SHA256}"
+EXTRA_OECMAKE = "-DCMAKE_BUILD_TYPE=Release -DAKTUALIZR_VERSION=${PV}"
+
+GARAGE_SIGN_OPS = "${@ '-DGARAGE_SIGN_VERSION=%s' % d.getVar('GARAGE_SIGN_VERSION') if d.getVar('GARAGE_SIGN_VERSION') is not None else ''} \
+                   ${@ '-DGARAGE_SIGN_SHA256=%s' % d.getVar('GARAGE_SIGN_SHA256') if d.getVar('GARAGE_SIGN_SHA256') is not None else ''} \
+                  "
+
+PACKAGECONFIG ?= "ostree ${@bb.utils.filter('DISTRO_FEATURES', 'systemd', d)} ${@bb.utils.filter('SOTA_CLIENT_FEATURES', 'hsm serialcan ubootenv', d)}"
+PACKAGECONFIG_class-native = "sota-tools"
+PACKAGECONFIG[warning-as-error] = "-DWARNING_AS_ERROR=ON,-DWARNING_AS_ERROR=OFF,"
+PACKAGECONFIG[pedantic-warnings] = "-DPEDANTIC_WARNINGS=ON,-DPEDANTIC_WARNINGS=OFF,"
+PACKAGECONFIG[code-coverage] = "-DBUILD_WITH_CODE_COVERAGE=ON,-DBUILD_WITH_CODE_COVERAGE=OFF,"
+PACKAGECONFIG[ostree] = "-DBUILD_OSTREE=ON,-DBUILD_OSTREE=OFF,ostree,"
+PACKAGECONFIG[deb] = "-DBUILD_DEB=ON,-DBUILD_DEB=OFF,dpkg,"
+PACKAGECONFIG[hsm] = "-DBUILD_P11=ON,-DBUILD_P11=OFF,libp11,"
+PACKAGECONFIG[sota-tools] = "-DBUILD_SOTA_TOOLS=ON ${GARAGE_SIGN_OPS},-DBUILD_SOTA_TOOLS=OFF,glib-2.0,"
+PACKAGECONFIG[opcua] = "-DBUILD_OPCUA=ON,-DBUILD_OPCUA=OFF,boost,"
+PACKAGECONFIG[isotp] = "-DBUILD_ISOTP=ON,-DBUILD_ISOTP=OFF,"
+PACKAGECONFIG[systemd] = "-DBUILD_SYSTEMD=ON,-DBUILD_SYSTEMD=OFF,systemd,"
+PACKAGECONFIG[load-tests] = "-DBUILD_LOAD_TESTS=ON,-DBUILD_LOAD_TESTS=OFF,"
+PACKAGECONFIG[fault-injection] = "-DFAULT_INJECTION=ON,-DFAULT_INJECTION=OFF,libfiu"
+PACKAGECONFIG[serialcan] = ",,,slcand-start"
+PACKAGECONFIG[ubootenv] = ",,,u-boot-fw-utils aktualizr-uboot-env-rollback"
```

Comment:
I wonder whether really having all of them as PACKAGECONFIG makes sense in reality (e.g. warning-as-error, pedantic-warnings or isotp, are those really things we want to have configurable through PACKAGECONFIG?). But I guess they don't hurt...

Reply:
Yes, they are not necessary, but I was trying to make the PACKAGECONFIG aligned with the options() in CMakeFIle.txt of aktualizr source code, :)

Reply:
Ok I see. I do not have a strong opinion on whether to add all or only the necessary ones.

@patrickvacek what is your thought on this?

Reply:
One bad side effect of having so many package config options is that DISTRO_FEATURES can end up having way too many options that might end up being specific to one single recipe (or generating unexpected side effects on other recipes that might be using the same string for a different purpose).